zilliqa/src/node_launcher.rs

```diff
@@ -290,7 +293,12 @@ impl NodeLauncher {
                 r = self.reset_timeout_receiver.next() => {
                     let sleep_time = r.expect("reset timeout stream should be infinite");
                     trace!(?sleep_time, "timeout reset");
-                    sleep.as_mut().reset(Instant::now() + sleep_time);
+                    consensus_sleep.as_mut().reset(Instant::now() + sleep_time);
+                },
+
+                () = &mut mempool_sleep => {
+                    self.node.lock().unwrap().process_transactions_to_broadcast()?;
+                    mempool_sleep.as_mut().reset(Instant::now() + Duration::from_millis(20));
```

Comment:
Is 20ms too regular? I don't know. Personally, I would prefer less regular but larger batches e.g. 40 batches of 100 txns rather than 100 batches of 40 txns. There is a cost to do the underlying gossip.

Reply:
Yea, my hunch is that it might work better. It may also have less routing overhead i.e. less big flows, rather than more small flows.